### DISCUSSION TITLE
Reminder for scaling prismatic joint values for PSM

### DISCUSSION THREAD

**Discussion by jackzhy96** [surgical-robotics-ai/surgical_robotics_challenge#73] [Ideas]
Hi All,

This is a kind reminder for the people who want to replay their recorded kinematic data. If you record the kinematic data from the ROS topic for AMBF PSM  (i.e. `/ambf/env/psm1(2)/baselink/State` ) and want to replay through `psm.servo_jp()`, please make sure that you manually divide the third joint of the PSM (the prismatic joint) by 10 before feeding into the `servo_jp()`. 

In addition, sometimes the gripper command may be not reliable because of some obstacles or blocking in the simulation scene. You may consider using an offset if necessary.

The original discussion is through here:
https://github.com/collaborative-robotics/surgical_robotics_challenge/issues/64#issuecomment-1507567145

**Comment by adnanmunawar** [comment]
Thanks for this discussion Jack. For context here is my reply:

>I would suggest that if you are recording the state (joint space or Cartesian space) of the PSMs, you should record them from the measured_jp topic that is available after running the [launch_crtk_interface.py](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/launch_crtk_interface.py) script. This script also launches a subscriber called servo_jp, and `move_cp`. Recording and publishing via these topics (interfaces) will allow you to not worry about the scale.
